infra/aws/org-bootstrap.md

```diff
@@ -0,0 +1,394 @@
+}
+```
+
+This code creates a new `infra` role in the `spacecats-infra` account and
+allows users in the `spacecats-id` account to assume that role in this
+account as well. Note that we're not gating this based on the role in the
+`id` account; instead, we assume the `id` account will do that gating for
+us. Then we grant the `infra` role the same level of permissions those
+users had in the `id` account. Note that we *could* give them a different
+level of access if we wanted to -- general engineers will not have
+permission to do much of anything in the `id` account most of the time,
+but we could grant them power user access in a sandbox account, and
+perhaps only access to look at Cloudwatch logs in the staging or prod
+accounts. Using this method allows you to do all your user management in
+one place, and then give role-based access everywhere else.
+
+In order for these users to access other accounts, you'll need to set up
+their profiles for these other accounts like so:
+
+```text
+[profile spacecats-infra]
+source_profile = spacecats-id
+role_arn = arn:aws:iam::<spacecats-infra account number>:role/infra
+region = us-west-2
+output = json
+# If you have MFA turned on, you will also need to add the mfa_serial
+# line from the source_profile.
+```
+
+## Next Steps
+
+Once you have the `org-root`, `id`, and `infra` accounts created, you
+can move on to setting up the other parts of your infrastructure. If you
+want to create more accounts, follow the example with the `infra` account
+above; you should refrain from creating any IAM users in non-`id` accounts
+if possible (though sometimes this is necessary for CI/CD or other
+automation).
+
+You should also set up some basic AWS services for the organization. Both
+Cloudtrail and GuardDuty can be set up at the organization level in the
+`org-root` account, and Config should be set up in each account.
+
+### Shared Resources
+
+In general, with an organization, there should be a bias towards making
+each individual account as self-sufficient as possible and avoid using
+too many shared resources. One of the driving reasons for using an AWS
+Organization is the ability to compartmentalize resources; engineers can
+have wide permissions in a sandbox account, for instance, because they
+are not going to be able to touch anything in the main release pipeline.
+If you use a shared resource, like an S3 bucket, you end up having to
+do more detailed permission management to keep accounts from stepping on
+each other -- this essentially eliminates the advantage of the
+organization in the first place.
+
+For most things, creating a separate resource for each account is likely
+the better play -- for things like S3 buckets, having two S3 buckets with
+half as much stuff as a single larger S3 bucket ends up essentially being
+the same cost. Even for things where the added cost is marginal, like
+having multiple ECR repositories, the added peace of mind keeping your
```

Comment:
I'm not sure multiple ECRs is the right example for exemplifying why it makes sense to isolate resources on a per account basis. While the additional costs may not be a big deal, the complexity and time this kind of design adds to a delivery pipeline is a negative tradeoff. I would personally lean towards putting the ECR in the infra account.

How about using examples like separating VPCs or ECS services to each account instead?